execution_workflows/APAtrap/bin/convert_to_tsv.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+
+import sys
+import argparse
+import pandas as pd
+
+def parse_args(args=None):
+    Description = "Reformat APAtrap deAPA output txt file into the output files of differential challenge"
+    Epilog = "Example usage: python convert_to_tsv.py <FILE_IN> <DIFFERENTIAL_OUT>"
+
+    parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+    parser.add_argument("FILE_IN", help="Input deAPA output txt file.")
+    parser.add_argument("DIFFERENTIAL_OUT", help="Name of output file for differential challenge")
+    return parser.parse_args(args)
+
+
+def reformat_bed(file_in, differential_out):
+    """
+    This function reformats the txt deAPA output file to files for
+    differential challenges
+    :param file_in: txt file to be reformatted
+    :return: N/A
+    """
+    differential_out = open(differential_out, "wt")
+
+    df = pd.read_csv(file_in, sep='\t')
+    for index, row in df.iterrows():
+        # write differential file
+        # keep just the gene name
+        name = row['Gene'].split("|")[0]
+        significance = str(row['p.value'])
+        output = [name, significance]
+        differential_out.write("\t".join(output) + "\n")
+
+    differential_out.close()
```

Comment:
When I ran the test data, I get transcript IDs in the first column (e.g. ENSMUSTxxxxx etc.), but I think the [specs want gene IDs](https://github.com/iRNA-COSI/APAeval/blob/main/execution_workflows/execution_output_specification.md#format-03).

I can see in APAtrap's output files that it does not retain the gene ID (ENSMUSGxxx)? I don't know if this will affect summary workflows for the differential usage challenges, but I suspect if we're wanting to match to 'ground truth' this would fail if the ground truth files have gene IDs. I wouldn't be sure how to resolve this either if so, as genes could have different transcript IDs with different p-values.

@ninsch3000 @yuukiiwa @mrgazzara any input on the differential usage ground truth TSVs and whether this is compatible with the summary workflows?

Reply:
@faricazjj - We discussed this in the meeting yesterday and decided that if a tool runs a statistical test transcript-by-transcript like here, we will pick the **smallest p-value for the corresponding gene**, and report the gene_id & p-value as according to the [specs](https://github.com/iRNA-COSI/APAeval/blob/main/execution_workflows/execution_output_specification.md#format-03). So I think you will need to do the following:

1. Generate a table mapping transcript_ids to gene_ids from the input GTF file (e.g. like the script in the [DaPars2 workflow here](execution_workflows/DaPars2/Dapars2_snakemake/workflow/scripts/extractIdSymbol.py))
2. Use above table to assign transcript_ids in output files to corresponding gene_id, then pick the smallest p-value for each gene & report those in the differential_usage TSV file.